core/src/main/scala/com/pingcap/tispark/utils/WriteUtil.scala

```diff
@@ -228,98 +228,42 @@ object WriteUtil {
       row: WrappedRow,
       index: TiIndexInfo,
       tiTable: TableCommon,
-      remove: Boolean): WrappedEncodedRow = {
-    if (index.isUnique) {
-      val (encodedKey, encodedValue) =
-        generateUniqueIndexKey(row.row, row.handle, index, tiTable, remove)
-      WrappedEncodedRow(
-        row.row,
-        row.handle,
-        encodedKey,
-        encodedValue,
-        isIndex = true,
-        index.getId,
-        remove)
-    } else {
-      val (encodedKey, encodedValue) =
-        generateSecondaryIndexKey(row.row, row.handle, index, tiTable, remove)
-      WrappedEncodedRow(
-        row.row,
-        row.handle,
-        encodedKey,
-        encodedValue,
-        isIndex = true,
-        index.getId,
-        remove)
-    }
+      remove: Boolean) = {
+    val (encodedKey, encodedValue) =
+      generateIndexKeyAndValue(row.row, row.handle, index, tiTable, remove)
+    WrappedEncodedRow(
+      row.row,
+      row.handle,
+      encodedKey,
+      encodedValue,
+      isIndex = true,
+      index.getId,
+      remove)
   }
 
   /**
    * construct unique index and non-unique index and value to be inserted into TiKV
    * NOTE:
-   * pk is not handle case is equivalent to unique index.
-   * for non-unique index, handle will be encoded as part of index key. In contrast, unique
-   * index encoded handle to value.
+   *      pk is not handle case is equivalent to unique index.
```

Comment:
Plz change the comment related to code change

Reply:
> Plz change the comment related to code change

The logic of this part code is not changed. The comment is no need to change.